src/mame/machine/tsconf.cpp

```diff
@@ -72,191 +74,381 @@ void tsconf_state::tsconf_update_bank1()
-							x++;
-						}
-						else
-						{
-							*bm++ = pix;
-						}
-						video_location++;
-					}
-				}
+				screen.max_y++;
+				draw = true;
 			}
+			if (draw)
+				m_previous_tsu_vpos = m_screen->vpos();
+		}
+
+		if (draw)
+		{
+			m_screen->priority().fill(0, screen);
+			if (BIT(m_regs[TS_CONFIG], 5))
+				m_ts_tilemap[TM_TILES0]->draw(*m_screen, m_screen->curbitmap().as_ind16(), screen,
+											  BIT(m_regs[TS_CONFIG], 2) ? TILEMAP_DRAW_ALL_CATEGORIES : TILEMAP_DRAW_CATEGORY(1), 1);
+
+			if (BIT(m_regs[TS_CONFIG], 6))
+				m_ts_tilemap[TM_TILES1]->draw(*m_screen, m_screen->curbitmap().as_ind16(), screen,
+											  BIT(m_regs[TS_CONFIG], 3) ? TILEMAP_DRAW_ALL_CATEGORIES : TILEMAP_DRAW_CATEGORY(1), 2);
+
+			if (BIT(m_regs[TS_CONFIG], 7))
+				draw_sprites(screen);
 		}
 	}
 }
 
+void tsconf_state::spectrum_UpdateBorderBitmap()
+{
+	spectrum_UpdateScreenBitmap();
+}
+
 u16 tsconf_state::get_border_color()
 {
 	return m_regs[BORDER];
 }
 
-void tsconf_state::draw_sprites(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect)
+/*
+SFILE	Reg.16	7		6		5		4		3		2		1		0
+0		R0L		Y[7:0]
+1		R0H		YF		LEAP	ACT		-		YS[2:0]					Y[8]
+2		R1L		X[7:0]
+3		R1H		XF		-		-		-		XS[2:0]					X[8]
+4		R2L		TNUM[7:0]
+5		R2H		SPAL[7:4]						TNUM[11:8]
+*/
+void tsconf_state::draw_sprites(const rectangle &cliprect)
 {
-	//u8 *messram = m_ram->pointer();
-	//u8 *sprites_location = messram + PAGE4K(m_regs[SG_PAGE]);
-	for (u8 i = 0; i < 85; i++)
+	rectangle screen = get_screen_area();
+
+	u8 layer = 0;
+	u8 *sinfo = m_sfile->pointer() + 1;
+	u8 sid = 1;
+	// Higher Sprite draws on top. Prepare to iterate backwards.
+	for (; sid < 85 && layer < 3; sid++)
 	{
-		logerror("Draw Sprites ... TODO\n");
+		if (BIT(*sinfo, 6))
+			layer++;
+		sinfo += 6;
+	}
+	sinfo -= 1;
+
+	for (; sid; sid--)
+	{
+		s16 y = *sinfo++;
+		y |= BIT(*sinfo, 0) << 8;
+		y += screen.top() - (y >= screen.height() ? 512 : 0);
+		u8 height8 = BIT(*sinfo, 1, 3);
+		layer -= BIT(*sinfo, 6);
+		if (!BIT(*sinfo, 5))
+		{
+			// sprite disabled -> move to previous
+			sinfo -= 7;
+		}
+		else
+		{
+			bool flipy = BIT(*sinfo++, 7);
+			s16 x = *sinfo++;
+			x |= BIT(*sinfo, 0) << 8;
+			x += screen.left() - (x >= screen.width() ? 512 : 0);
+			u8 width8 = BIT(*sinfo, 1, 3);
+			bool flipx = BIT(*sinfo++, 7);
+			u16 code = *sinfo++;
+			code |= BIT(*sinfo, 0, 4) << 8;
+			u8 pal = BIT(*sinfo, 4, 4);
+			sinfo -= 11;
+
+			u32 pmask = layer ? (layer == 1 ? GFX_PMASK_2 : 0) : (GFX_PMASK_1 | GFX_PMASK_2);
+
+			u8 tile_row = code / 64 + flipy * height8;
+			for (auto iy = y; iy <= y + height8 * 8; iy = iy + 8)
+			{
+				u8 tile_col = (code % 64) + flipx * width8;
+				for (auto ix = x; ix <= x + width8 * 8; ix = ix + 8)
+				{
+					m_gfxdecode->gfx(TM_SPRITES)->prio_transpen(m_screen->curbitmap().as_ind16(), cliprect, tmp_tile_oversized_to_code((tile_row % 64) * 64 + (tile_col % 64)), pal, flipx, flipy, ix, iy, m_screen->priority(), pmask, 0);
```

Comment:
That is also drawback of oversized tiles. [64x64x8]*8 map easily overflows u16 which is max accepted "code" in prio_transpen